app/src/main/java/com/android/shelfLife/ui/newoverview/NewHouseHoldCreationScreen.kt

```diff
@@ -48,45 +40,33 @@ import kotlinx.coroutines.launch
 @Composable
 fun HouseHoldCreationScreen(
     navigationActions: NavigationActions,
-    houseHoldRepository: HouseHoldRepository,
-    invitationRepository: InvitationRepository,
-    userRepository: UserRepository
+    viewModel: HouseholdCreationScreenViewModel = hiltViewModel()
 ) {
-  val householdCreationScreenViewModel: HouseholdCreationScreenViewModel = viewModel {
-    HouseholdCreationScreenViewModel(
-        houseHoldRepository = houseHoldRepository,
-        invitationRepository = invitationRepository,
-        userRepository = userRepository)
-  }
-  val householdToEdit by householdCreationScreenViewModel.householdToEdit.collectAsState()
+  val householdToEdit by viewModel.householdToEdit.collectAsState()
 
   var isError by rememberSaveable { mutableStateOf(false) }
   var houseHoldName by rememberSaveable { mutableStateOf(householdToEdit?.name ?: "") }
 
   var showConfirmationDialog by rememberSaveable { mutableStateOf(false) }
 
-  // Mutable state list to hold member emails
-  val memberEmailList by householdCreationScreenViewModel.emailList.collectAsState()
+  val memberEmailList by viewModel.emailList.collectAsState()
   var emailInput by rememberSaveable { mutableStateOf("") }
   var showEmailTextField by rememberSaveable { mutableStateOf(false) }
 
   val coroutineScope = rememberCoroutineScope()
   val columnScrollState = rememberScrollState()
-
   val focusRequester = remember { FocusRequester() }
 
-  // Scroll to the bottom and focus on the email input field when the email input field is shown
   LaunchedEffect(showEmailTextField) {
     if (showEmailTextField) {
       coroutineScope.launch { columnScrollState.animateScrollTo(columnScrollState.maxValue) }
       focusRequester.requestFocus()
     }
   }
 
-  // Function to add email card to the list and scroll to the bottom
   fun addEmailCard() {
-    if (emailInput.isNotBlank() && emailInput.trim() !in memberEmailList) {
-      householdCreationScreenViewModel.addEmail(emailInput.trim())
+    val added = viewModel.tryAddEmailCard(emailInput)
```

Comment:
This is a good simiplication such that the logic is called in a function within the viewModel